client/blocks/app-banner/index.jsx

```diff
@@ -0,0 +1,148 @@
+ */
+import Button from 'components/button';
+import Card from 'components/card';
+import { getActionLog } from 'state/ui/action-log/selectors';
+import {
+	getAppBannerData,
+	getPageType,
+	isDismissed,
+	getNewDismissTimes,
+	ALLOWED_PAGE_TYPES
+} from './utils';
+import {
+	findLast,
+	get,
+	identity,
+	includes,
+	noop
+} from 'lodash';
+import { isMobile } from 'lib/viewport';
+import { getUserSetting, isNotificationsOpen } from 'state/selectors';
+import { recordTracksEvent } from 'state/analytics/actions';
+import { ROUTE_SET } from 'state/action-types';
+import { saveUserSettings } from 'state/user-settings/actions';
+
+class AppBanner extends Component {
+	static propTypes = {
+		translate: PropTypes.func,
+		trackAppBannerImpression: PropTypes.func,
+		trackAppBannerDismiss: PropTypes.func,
+		trackAppBannerOpen: PropTypes.func,
+		saveDismissTime: PropTypes.func,
+		// connected
+		pageType: React.PropTypes.string,
+		dismissedUntil: React.PropTypes.object,
+	};
+
+	static defaultProps = {
+		translate: identity,
+		trackAppBannerImpression: noop,
+		trackAppBannerDismiss: noop,
+		trackAppBannerOpen: noop,
+		saveDismissTime: noop,
+		dismissedUntil: {}
+	};
+
+	state = {
+		isHidden: false,
+	};
+
+	isVisible() {
+		const { pageType, dismissedUntil } = this.props;
+
+		return isMobile() && ! isDismissed( dismissedUntil, pageType ) && ! this.state.isHidden;
+	}
+
+	dismiss = ( event ) => {
+		event.preventDefault();
+		const { pageType } = this.props;
+
+		this.setState( { isHidden: true } );
+		this.props.saveDismissTime( pageType );
+		this.props.trackAppBannerDismiss( pageType );
+	};
+
+	openApp = ( event ) => {
+		event.preventDefault();
+		const { pageType } = this.props;
+
+		this.props.trackAppBannerOpen( pageType );
+	};
+
+	render() {
+		const { translate, pageType } = this.props;
+
+		if ( ! includes( ALLOWED_PAGE_TYPES, pageType ) ) {
+			return null;
+		}
+
+		if ( ! this.isVisible() ) {
+			return null;
+		}
+
+		const { title, copy } = getAppBannerData( translate, pageType );
+
+		this.props.trackAppBannerImpression( pageType );
+
+		// TODO: generate deep links
+		const deepLink = '#';
+
+		return (
+			<Card className={ classNames( 'app-banner', 'is-compact', pageType ) }>
+				<div className="app-banner__text-content">
+					<div className="app-banner__title">
+						<span> { title } </span>
+					</div>
+					<div className="app-banner__copy">
+						<span> { copy } </span>
+					</div>
+				</div>
+				<div className="app-banner__buttons">
+					<Button
+						className="app-banner__open-button"
+						onClick={ this.openApp }
+						href={ deepLink }
+					>
+						{ translate( 'Open in app' ) }
+					</Button>
+					<a
+						className="app-banner__no-thanks-button"
+						onClick={ this.dismiss }
+					>
+						{ translate( 'No thanks' ) }
+					</a>
+				</div>
+			</Card>
+		);
+	}
+}
+
+const mapStateToProps = ( state ) => {
+	const currentRoute = findLast( getActionLog( state ), { type: ROUTE_SET } );
```

Comment:
We may not need this, but if you find that you need this, I wonder if it's better to either add a ui reducer for the lastRoute, or memoize the selector in the global selector folder.